packages/narwallets/src/lib/narwallets.ts

```diff
@@ -0,0 +1,274 @@
+const Narwallets: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  metadata,
+  store,
+  provider,
+  emitter,
+  logger,
+}) => {
+
+  const signOut = async () => {
+    if (!(await isSignedIn())) {
+      return;
+    }
+
+    const res = await sendToNarwallets("sign-out");
+    // const res = await _state.wallet.signOut();
+    if (res === true) {
+      return;
+    }
+
+    const error = new Error(
+      typeof res.error === "string" ? res.error : res.error.type
+    );
+
+    // Prevent signing out by throwing.
+    if (error.message === "User reject") {
+      throw error;
+    }
+
+    // Continue signing out but log the issue.
+    logger.log("Failed to sign out");
+    logger.error(error);
+  };
+
+  return {
+    async signIn() {
+      const isUserSignedIn = await isSignedIn();
+      let code;
+      if (!isUserSignedIn) {
+        code = NARWALLETS_CODES.SIGN_IN;
+      } else {
+        code = NARWALLETS_CODES.GET_ACCOUNT_ID;
+      }
+      const response = await sendToNarwallets(code);
+      return [{ accountId: response }];
+    },
+
+    signOut,
+
+    async getAccounts(): Promise<Array<Account>> {
+      const accountId = await getAccountId();
+      return [{ accountId }];
+    },
+
+    async verifyOwner({ message }) {
+      logger.log("Narwallets:verifyOwner", { message });
+      throw new Error(`Method not supported by ${metadata.name}`);
+    },
+
+    async signAndSendTransaction({ signerId, receiverId, actions }) {
+      logger.log("signAndSendTransaction", { signerId, receiverId, actions });
+
+      const { contract, accounts } = store.getState();
+
+      if (!accounts || accounts.length == 0 || !contract) {
+        throw new Error("Wallet not signed in");
+      }
+
+      return callSignAndSendTransaction({
+        signerId,
+        receiverId: receiverId || contract.contractId,
+        actions: actions
+      })
+    },
+
+    async signAndSendTransactions({ transactions }) {
+      logger.log("signAndSendTransactions", { transactions });
+
+      const { contract, accounts } = store.getState();
+      
+      // test: avoid a call to isSignedIn
+      if (!accounts || accounts.length == 0 || !contract) {
+        throw new Error("Wallet not signed in");
+      }
+
+      return callSignAndSendTransactions(  
+        transactions
+      )
+    },
+  };
+};
+
+export function setupNarwallets({
+  iconUrl = "./assets/narwallets-logo.png",
+}: NarwalletsParams = {}): WalletModuleFactory<InjectedWallet> {
+  return async () => {
+
+    const mobile = isMobile();
+    if (mobile) {
+      return null;
+    }
+
+    setupNarwalletsState();
+
+    const installed = await isInstalled();
+
+    return {
+      id: "narwallets",
+      type: "injected",
+      metadata: {
+        name: "Narwallets",
+        description: null,
+        iconUrl,
+        downloadUrl:
+          "https://chrome.google.com/webstore/detail/narwallets-v4/lkpeokpdkmcdaiadpmnnpimlgmdobkdj",
+        deprecated: false,
```

Comment:
Can you make the deprecated configurable form the `setupNarwallets` you might see how it's done in other wallets?

Reply:
Done